opennms-bootstrap/src/main/java/org/opennms/bootstrap/FilesystemPermissionValidator.java

```diff
@@ -132,25 +131,35 @@ private void validatePath(final String user, final Path dirPath) throws Filesyst
             }
         }
 
-        final List<Path> failures;
-        try (final Stream<Path> stream = Files.walk(dirPath, FileVisitOption.FOLLOW_LINKS)) {
-            failures = stream
-                    .filter(file -> {
-                        final var uri = file.toUri().toString();
-                        if (uri.contains("/.git/")) return false;
-                        if (uri.contains("/lost+found")) return false;
-                        return true;
-                    }).filter(file -> {
+        final List<Path> failures = new ArrayList<>();
+        try {
+            final var iter = Files.walk(dirPath, FileVisitOption.FOLLOW_LINKS).iterator();
+            while (true) {
+                try {
+                    if (iter.hasNext()) {
```

Comment:
are there cases where iter.hasNext() can throw unexpected exceptions here? If not, that can replace `while(true)`

Reply:
From what I had read, if the file or directory exists but is unreadable due to permissions, it can throw a runtime exception in `hasNext`, but I didn't validate that directly.

Reply:
And actually, I re-evaluated what I did and I found some thinkos in other parts of the logic, so I ended up redoing a decent amount and I'm much happier with it now.  Mind taking another look?